### PR TITLE
Fix local development

### PR DESCRIPTION
**Pull request by Mandur**
# PR to fix local development

## What is being addressed

With the addition of the cloud deployment, the local development experience always starts with the Edge mode. The proposed fix explicitly disable edge when running in debug mode.

### CODE REVIEW

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWanNetworkSrvModule/Program.cs [hunk 1]
```diff
@@ -32,4 +32,6 @@
 };
 
 var configuration = NetworkServerConfiguration.CreateFromEnvironmentVariables();
+// when running in local debug mode we explicitely disable IoT Edge
+configuration.RunningAsIoTEdgeModule = false;
```
Comment (danigian) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWanNetworkSrvModule/Program.cs:36]: I don't get the fix. This is not only applied for local development, but also to any docker image being built no?

I would rather leave the code as is, but add something in the docs for the local development

Comment (Mandur) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWanNetworkSrvModule/Program.cs:36]: Woops. you are totally right, I was remembering this project was only for local dev and I was obviously wrong. Thanks for the catch.

What do you think to execute this line if we have a debugger attached? This typically should not occur in normal scenarios?

Comment (danigian) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWanNetworkSrvModule/Program.cs]: I would personally prefer to have cleaner code, therefore I would go for completely removing it and just adding a documentation note. As an alternative we might make use of launchsettings.json and appsettings.development.json for any dev related parameter like this

Comment (Mandur) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWanNetworkSrvModule/Program.cs]: I added a new parameters 'LOCAL_DEVELOPMENT' that will disable edge when set to true, If agreed I will also edit the [docs](https://azure.github.io/iotedge-lorawan-starterkit/2.1.0/user-guide/devguide/#debugging-in-visual-studio-outside-of-iot-edge-and-docker)

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs [hunk 2]
```diff
@@ -154,7 +156,7 @@ public static NetworkServerConfiguration CreateFromEnvironmentVariables()
             var envVars = new CaseInsensitiveEnvironmentVariables(Environment.GetEnvironmentVariables());
             config.ProcessingDelayInMilliseconds = envVars.GetEnvVar("PROCESSING_DELAY_IN_MS", config.ProcessingDelayInMilliseconds);
             config.RunningAsIoTEdgeModule = !envVars.GetEnvVar("CLOUD_DEPLOYMENT", false);
```
Comment (danigian) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs:158]: We can make this row check both CLOUD_DEPLOYMENT and LOCAL_DEVELOPMENT (if any of them is true, the RunningAsIoTEdgeModule will be false). In this way we don't need to alter the Program.cs

Let's also update the docs (please create an issue) with the steps to eventually start a Redis instance for testing those functionalities that require it

**Review by danigian — CHANGES_REQUESTED** [review]

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWanNetworkSrvModule/Program.cs (hunk 1 above)
Comment (danigian) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWanNetworkSrvModule/Program.cs]: Replied in another comment. Let's move the discussion there. It's fine to have local_development but we can simplify code

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWanNetworkSrvModule/Program.cs [hunk 3]
```diff
@@ -32,4 +32,5 @@
 };
 
 var configuration = NetworkServerConfiguration.CreateFromEnvironmentVariables();
+
```
Comment (danigian) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWanNetworkSrvModule/Program.cs:35]:
```suggestion
```
Let's remove the added whiteline here 👍

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs [hunk 4]
```diff
@@ -145,6 +145,8 @@ public class NetworkServerConfiguration
         /// </summary>
         public int ProcessingDelayInMilliseconds { get; set; } = Constants.DefaultProcessingDelayInMilliseconds;
 
+        public bool IsLocalDevelopment { get; set; }
```
Comment (danigian) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs:148]: I would say that if we don't use it, we should remove it and eventually add later on.
But it's also fine if you want to keep here and use it in the future

**Review by danigian — APPROVED** [review]

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs (hunk 4 above)
Comment (Mandur) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs]: Done

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs (hunk 2 above)
Comment (Mandur) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs]: Done

**Pull request merged**